spec/acceptance/resources_controller_spec.rb

```diff
@@ -9,6 +9,421 @@
   let(:raw_post) { params.to_json }
   let(:authorization) { AuthToken.generic_token }
 
+  let(:tool_group_one) { FactoryBot.create(:tool_group, name: "one") }
+
+  let(:resource_1) { Resource.find(1) }
+  let(:resource_2) { Resource.find(2) }
+  let(:resource_3) { Resource.find(3) }
+  let(:resource_4) { Resource.find(4) }
+  let(:resource_5) { Resource.find(5) }
+
+  let(:languages_fr_en) { ["fr", "en"] }
+  let(:languages_fr_it) { ["fr", "it"] }
+  let(:languages_fr_es) { ["fr", "es"] }
+  let(:languages_fr) { ["fr"] }
+  let(:languages_it) { ["it"] }
+  let(:languages_en) { ["en"] }
+  let(:languages_es) { ["es"] }
+
+  let(:countries_fr) { ["FR"] }
+  let(:countries_gb) { ["GB"] }
+  let(:countries_nz) { ["NZ"] }
+  let(:countries_fr_us) { ["FR", "US"] }
+  let(:countries_gb_us_nz) { ["GB", "US", "NZ"] }
+
+  let(:openness_1) { [1] }
+  let(:openness_2) { [2] }
+  let(:openness_3) { [3] }
+  let(:openness_1_2) { [1, 2] }
+  let(:openness_2_3) { [2, 3] }
+  let(:openness_3_4) { [3, 4] }
+
+  let(:confidence_1) { [1] }
+  let(:confidence_2) { [2] }
+  let(:confidence_3) { [3] }
+  let(:confidence_1_2) { [1, 2] }
+  let(:confidence_2_3) { [2, 3] }
+  let(:confidence_3_4) { [3, 4] }
+
+  get "resources/suggestions" do
+    before(:each) do
+      # FactoryBot.create(:rule_country, tool_group: tool_group_one, countries: countries_fr_us)
+      FactoryBot.create(:rule_language, tool_group: tool_group_one, languages: languages_fr_en)
+      # FactoryBot.create(:rule_praxis, tool_group: tool_group_one, openness: openness_1, confidence: confidence_2)
+
+      ResourceToolGroup.create!(resource_id: resource_1.id, tool_group_id: tool_group_one.id, suggestions_weight: 2.0)
+      ResourceToolGroup.create!(resource_id: resource_2.id, tool_group_id: tool_group_one.id, suggestions_weight: 1.5)
+      ResourceToolGroup.create!(resource_id: resource_3.id, tool_group_id: tool_group_one.id, suggestions_weight: 1.7)
+      ResourceToolGroup.create!(resource_id: resource_4.id, tool_group_id: tool_group_one.id, suggestions_weight: 3.0)
+      ResourceToolGroup.create!(resource_id: resource_5.id, tool_group_id: tool_group_one.id, suggestions_weight: 1.0)
+    end
+
+    context "when matching a tool group with only language rule" do
+      context "if matching none of languages defined in rule" do
+        it "does not return coincidences" do
+          do_request "filter[language]": languages_es
+
+          expect(status).to be(200)
+          expect(JSON.parse(response_body)["data"].count).to eql 0
+        end
+      end
+
+      context "if matching any of the languages defined in rule" do
+        it "return coincidences" do
+          do_request "filter[language]": languages_fr_it
+
+          expect(status).to be(200)
+          expect(JSON.parse(response_body)["data"].count).to eql 5
```

Comment:
can the coincidences be checked that they're the correct ones and not just that there's 5?

Reply:
I'm actually kind of surprise and confused to see the term "coincidence" -- which is defined as "a remarkable concurrence of events or circumstances without apparent causal connection." -- is that what was intended? Or is this maybe "co" "incidence", like a match? overlap? maybe a better term would be less confusing?

Nowhere in the code above is the term "coincidence" used.

Reply:
"coincident" means "occurring together in space or time" - perhaps that is what was meant - or "coinciding"

Reply:
@andrewroth @knutsenm 

I made changes here, when you ready take a look if this looks better now.